src/libraries/Common/src/Interop/Unix/System.Native/Interop.DeviceFiles.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    // mknod: https://man7.org/linux/man-pages/man2/mknod.2.html
+    // makedev, major and minor: https://man7.org/linux/man-pages/man3/makedev.3.html
+    internal static partial class Sys
+    {
+        internal static int CreateBlockDevice(string pathName, int mode, uint major, uint minor)
+        {
+            return MkNod(pathName, mode | FileTypes.S_IFBLK, major, minor);
+        }
+
+        internal static int CreateCharacterDevice(string pathName, int mode, uint major, uint minor)
+        {
+            return MkNod(pathName, mode | FileTypes.S_IFCHR, major, minor);
+        }
+
+        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_MkNod", StringMarshalling = StringMarshalling.Utf8, SetLastError = true)]
+        internal static partial int MkNod(string pathName, int mode, uint major, uint minor);
```

Comment:
```suggestion
        private static partial int MkNod(string pathName, int mode, uint major, uint minor);
```

I don't believe this is called outside of this class and can be made private.